File: scripts/record_video_client.py
#!/usr/bin/env python
import rospy
from streaming_media.srv import *

def main():
    rospy.init_node('record_video_client')

    try:
        rospy.wait_for_service("/record_video")
        client = rospy.ServiceProxy('/record_video', RecordVideo)

    
        res = client(True)
        rospy.loginfo("Video capture: %s" % res.result)

    except rospy.ServiceException, e:
       rospy.loginfo("Service call failed: %s" % e)

if __name__ == '__main__':
    main()

File: src/record_video_client.cpp
#include <ros/ros.h>
#include "streaming_media/RecordVideo.h"  // streaming_media 包名

int main(int argc, char** argv) {
    ros::init(argc, argv, "record_video_client");
    ros::NodeHandle nh;
    ros::ServiceClient client = nh.serviceClient<streaming_media::RecordVideo>("/record_video");
    streaming_media::RecordVideo srv;

    srv.request.record = true;  // Set to true to record video
    if (client.call(srv)) {
        ROS_INFO("Video capture: %s", srv.response.result.c_str());
    } else {
        ROS_ERROR("Failed to call service /record_video");
    }

    return 0;
}
